sycl/test-e2e/XPTI/buffer/in_cycle.cpp

```diff
@@ -66,7 +66,7 @@ int main() {
   // CHECK:{{[0-9]+}}|Release buffer|[[USERID3]]|[[BEID3]]
   // CHECK:{{[0-9]+}}|Destruct buffer|[[USERID3]]
   for (int i = 0; i < 3; i++)
-    MismatchFound &= func(Queue);
+    MismatchFound |= func(Queue);
```

Comment:
IMO, a better fix is to use `+=` if it returns number of mismatches found, or change the function to return `true/false` for success/failure.

Reply:
I didn't mean to change that to an int, have restored it back to it's original boolean declaration.

I'm disinclined to make any further change. The variable is a boolean named `MismatchFound`.  It's pretty clear what it is. If I were change it to be something that could default to true, it'd have to be `MismatchNotFound` and then I'd also have to change the `func(...)` that is called to similarly reverse its interpretation. 
It's fine having a boolean initialized to false, it just means the test has to use `|=` instead of `&=`  .